### PR TITLE
ZCS-3487: Added the test cases to verify the url value set at the global config

### PR DESCRIPTION
**Pull request by miteshsavani**

### CODE REVIEW

File: src/java/com/zimbra/qa/selenium/projects/admin/tests/domains/ZimbraHelpAdminURLModification.java [hunk 1]
```diff
@@ -107,5 +105,69 @@ public void ZimbraHelpAdminURLModification_01() throws HarnessException {
...
+
+	}
+	
+	@Bugs(ids="101023")
+	@Test( description = "Verify that zimbra admin help page is opened as per the value set in attribute ZimbraHelpAdminURL at the global config",
+			groups = { "functional", "L2" })
+
+	public void ZimbraHelpAdminURLModification_02() throws HarnessException {
+
+		try{
```
Comment (jiteshsojitra) [inline src/java/com/zimbra/qa/selenium/projects/admin/tests/domains/ZimbraHelpAdminURLModification.java]: Entire code should be left aligned.

**Review by swapnilpingle89 — APPROVED** [review]

File: src/java/com/zimbra/qa/selenium/projects/ajax/tests/main/attributes/ZimbraHelpAdvancedURL.java [hunk 2]
```diff
@@ -126,6 +126,60 @@ public void ZimbraHelpAdvancedURL_01() throws HarnessException {
 			}
 
 		}
+	}
+		@Bugs(ids = "101023")
+		@Test(description = "Verify the product help URL as per the value set in attribute ZimbraHelpAdminURL at the global config", priority=5,
+			groups = { "functional", "L3" })
+		public void ZimbraHelpAdvancedURL_02() throws HarnessException {
```
Comment (jiteshsojitra) [inline src/java/com/zimbra/qa/selenium/projects/ajax/tests/main/attributes/ZimbraHelpAdvancedURL.java]: Put one blank line between @test and public test method.

File: src/java/com/zimbra/qa/selenium/projects/ajax/tests/main/attributes/ZimbraHelpAdvancedURL.java [hunk 3]
```diff
@@ -126,6 +126,60 @@ public void ZimbraHelpAdvancedURL_01() throws HarnessException {
...
+			try {
+				// Modify the config and change the help URL
+				ZimbraAdminAccount.AdminConsoleAdmin()
+								.soapSend("<ModifyConfigRequest xmlns='urn:zimbraAdmin'>"
+								+ "<a n='zimbraHelpAdvancedURL'>" + url + "</a>"
+								+ "</ModifyConfigRequest>");
+
+				app.zPageMain.zRefreshMainUI();
+				app.zPageMain.zToolbarPressPulldown(Button.B_ACCOUNT, Button.O_PRODUCT_HELP);
+				SleepUtil.sleepMedium();
```
Comment (jiteshsojitra) [inline src/java/com/zimbra/qa/selenium/projects/ajax/tests/main/attributes/ZimbraHelpAdvancedURL.java]: Please keep all sleep in the function if possible.

File: src/java/com/zimbra/qa/selenium/projects/ajax/tests/main/attributes/ZimbraHelpAdvancedURL.java [hunk 4]
```diff
@@ -126,6 +126,60 @@ public void ZimbraHelpAdvancedURL_01() throws HarnessException {
...
+						app.zPageMain.sSelectWindow(id);
+
+						if (app.zPageMain.sGetTitle().contains("BBC")) {
+							//Get the opened URL
+							tempURL=app.zPageMain.sGetLocation();
+							found = true;
+							app.zPageMain.zSeparateWindowClose(app.zPageMain.sGetTitle());
+							break;
+						} else if (!(app.zPageMain.sGetTitle().contains("Zimbra: Inbox"))) {
+									app.zPageMain.zSeparateWindowClose(app.zPageMain.sGetTitle());
```
Comment (jiteshsojitra) [inline src/java/com/zimbra/qa/selenium/projects/ajax/tests/main/attributes/ZimbraHelpAdvancedURL.java]: This line is poorly formatted.

File: src/java/com/zimbra/qa/selenium/projects/ajax/tests/main/attributes/ZimbraHelpAdvancedURL.java [hunk 5]
```diff
@@ -126,6 +126,60 @@ public void ZimbraHelpAdvancedURL_01() throws HarnessException {
...
+						}
+					}
+					if (!found) {
+						tempURL=app.zPageMain.sGetLocation();
+					}
+				} else {
+					tempURL=app.zPageMain.sGetLocation();
```
Comment (jiteshsojitra) [inline src/java/com/zimbra/qa/selenium/projects/ajax/tests/main/attributes/ZimbraHelpAdvancedURL.java]: tempURL=app.zPageMain.sGetLocation(); should be: tempURL = app.zPageMain.sGetLocation();

**Review by jiteshsojitra — CHANGES_REQUESTED** [review]

**Comment by jiteshsojitra** [comment]
Thanks Mitesh for making these changes, now any new work/maintenance changes should have best formatted testcases like we discussed.

**Review by jiteshsojitra — APPROVED** [review]

**Review by singhjitesh — APPROVED** [review]

**Pull request closed without merging**